src/lib/common/Exo3LineParser.cpp

```diff
@@ -5,6 +5,10 @@
 #include "Exo3LineParser.h"
 #include "string.h"
 
+#ifndef __clang__
+#pragma GCC diagnostic ignored "-Wstringop-truncation"
+#endif
```

Comment:
Could you explain this one?

Reply:
This was failing on our builds. We currently are not building the SDI12 app (which we probably should and I will add that to another commit), but I cannot see how that build was passing due to this error failing compilation.
This warning is something that is not very serious as long as you are aware of what you are doing (which looking at the logic it makes sense), see [this](https://stackoverflow.com/questions/50198319/gcc-8-wstringop-truncation-what-is-the-good-practice) for a rundown.

This was occurring at line 44, because we assign token_len at line 34.

Reply:
I will dig in deeper to find out if this was building on develop and why.

Reply:
Nice. Thanks. I think that the warning is valuable and that we should fix the code rather than suppress the warning.

It's interesting that the compiler can't connect the dots to see that the final copied character will always be a null terminator. We can add insurance and help the compiler by inserting:

    token_copy[token_len + 1] = '\0';

immediately after the strncpy. I think then the warning should go away.

Reply:
Addressed this issue with a memcpy here.
The way that the `strncpy` was intended to be originally used was as a `memcpy` as `strncpy` will append a zero to the destination if the source is less than the length parameter.